### DISCUSSION TITLE
Show a pie chart with "unique" values from a database-table-row

### DISCUSSION THREAD

**Discussion by iwanhoogendoorn** [Budibase/budibase#11093] [Help] [answered]
Hi, 

I have a simple database and this database has a table and this table has a column named "IP addresses"
There are multiple records inside the table and some IP addresses are duplicates.

I want to have a simple pie chart or card where I display the IP addresses that are in that table but not display ALL IP addresses, but only take the unique ones out.

I started with a Data Provider, then a Repeater.

But I have no clue how I filter in a way that I only see the unique values...

**Comment by R2bEEaton** [comment, marked as answer]
Hello! As far as I'm aware there's no one-click solution in Budibase for getting unique records, but depending on what you need there are some options.

Some things to explore:

### Table views

If all you need is the unique IP addresses, you can do this with a table view in Budibase.

For example, if your data is set up like this, press Add View.

![image](https://github.com/Budibase/budibase/assets/34921506/ad7641b4-f7b8-4422-a1bd-6e7fb65920fd)

Then Calculate the count of some unique column.

![image](https://github.com/Budibase/budibase/assets/34921506/ab96be77-da7b-454b-a270-60c020a88d67)
![image](https://github.com/Budibase/budibase/assets/34921506/d383bae8-05ad-4866-9174-e4ab49e34894)

And then, crucially, group by IP.

![image](https://github.com/Budibase/budibase/assets/34921506/5868e42a-1dba-46ff-a1c4-1f4015a3c7c4)
![image](https://github.com/Budibase/budibase/assets/34921506/de26f187-3386-4cf7-bf7b-f1a936bfe2bb)

Now as you can see, you have access to all the unique IP addresses, and the number of rows associated with each.

![image](https://github.com/Budibase/budibase/assets/34921506/76d05553-1c8f-4482-bbe2-ddcad173ff5a)

You can use this View as a data source and use the Group column in your Pie chart. But of course, all the other data will be gone. If you want to access the other data, try the next solution if your data allows it.

### Using a more traditional RDBMS
If your data is in MySQL for example, you could create a custom query in Budibase and get records by unique IP address using `SELECT IP, Version, Owner FROM table GROUP BY (IP);` or [other examples here](https://stackoverflow.com/a/5967209). However which row you get is not defined by default, so if you want to get something like the most recent row you'll have to make a more fancy query.

### Hope this helps!
Let me know if you have any questions! If this works, make sure to accept this answer to help future users 😃

**Reply by melohagan** [reply]
If using SQL you can also use the [DISTINCT keyword](https://www.w3schools.com/Sql/sql_distinct.asp) in a [custom query](https://docs.budibase.com/docs/data-sources#custom-queries)

**Reply by dauphinpasdroit** [reply]
Hello everyone,

Is there a way to do the same using the new view feature ? 

I havn't updated my self host budibase yet, I tried on an online instance. I used your tutorial to make pie chart the same way as the OP but I couldn't performed the same in the new view feature...

**Reply by Sterh20** [reply]
I have the same problem as @alphaxr6. How to create a table of unique values?
